classes/Gems/Acl/Privilege.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Gems\Acl;
+
+use Gems\Exception\AuthenticationException;
+
+class Privilege
+{
+    public const SEPARATOR = '.';
+
+    public function __construct(
+        public readonly string $name,
+        public readonly ?string $label = null,
+        public readonly ?array $methods = null,
+    )
+    {}
+
+    public function getName(?string $method = null): string
+    {
+        if ($method !== null && $this->methods !== null) {
+            if (!in_array($method, $this->methods)) {
+                throw new AuthenticationException('Method not allowed');
+            }
+            return $this->name . self::SEPARATOR . $method;
+        }
+        return $this->name;
```

Comment:
Als `$this->methods` niet `null` is, is `$this->name` dus gelijk aan één van de methods, een soort duplicaat? Is het te overwegen om `$method` verplicht te maken als `$this->methods` niet `null` is?